pkg/client/kubernetes/scaling.go

```diff
@@ -70,3 +72,28 @@ func scaleResource(ctx context.Context, c client.Client, obj client.Object, repl
 	// see: https://github.com/kubernetes-sigs/controller-runtime/issues/172
 	return c.Patch(ctx, obj, client.RawPatch(types.MergePatchType, patch))
 }
+
+// WaitUntilDeploymentScaledToDesiredReplicas waits for the number of available replicas to be equal to the deployment's desired replicas count.
+func WaitUntilDeploymentScaledToDesiredReplicas(ctx context.Context, client client.Client, key types.NamespacedName, desiredReplicas int32) error {
+	return retry.UntilTimeout(ctx, 5*time.Second, 300*time.Second, func(ctx context.Context) (done bool, err error) {
+		deployment := &appsv1.Deployment{}
+		if err := client.Get(ctx, key, deployment); err != nil {
```

Comment:
What if the deployment is not found? We handle that case in the scaleMachineControllerManager, but not here.

Reply:
Hm, but why in this function? Isn't it odd that we return "OK" if we want to scale a deployment to 5 replicas and then it suddenly disappears?
I think this behaviour is highly specific to the restoration flow, so let's move this graceful "not found" handling out from here to the `Restore` function.
Sorry @kris94 for the back-and-forth :(

Reply:
Yes, moving it out makes sense.